Refactor the combined search implementation with form and search adapters

nikomakela opened the pull request:
LIIKUNTA-425.

### Combined search

The search page has 1 form, but that form triggers searches to multiple datasources:

1. The venue search tab makes a query to the Unified-Search.
2. The events and the course searches makes a query to the LinkedEvents.

This means that:

1. Parameters given as an input to the form should adapt to search from different datasources. 1 input can be called “q“ in another datasource, but “text“ in another.
2. Not all the parameters are used in every search, because not all the datasources have an equivalent filter.

#### Input: The URL (search) parameters

_The URL is the primary source for input and state._

The URL controls the whole search form, for easy maintain and also for the accessibility reasons:

1. The URL should be updated when new filters are applied (or any of them are removed / cleared). The active filters are always shown in the UI.
2. When another search tab is clicked, the search URL should not change from the search params.
3. When a filter is activated, the URL should be updated. When an URL is updated, the form should also ract to that.

An example of CombinedSearchAdapterInput type, that could be used as an input for the searches:

```typescript
// NOTE: this is just an example, not an actual implementation
type CombinedSearchAdapterInput = {
  text: string;
  orderBy: string;
  sportsCategories: string[];
  organization: string | null;
  keywords: string[];
};
```

The input URL should also be cleaned for the combined search. A conversion map can handle any needed _backward support_, but also to full fill any possible _mandatory inputs_, that should always exists there. It can also _initialize the search form_. It could convert some params to some others and initialize the form with the transformed keys and values.

E.g. `/search?order_by=field` could be transformed to `/search?orderBy=field&searchType=venue`.

#### Transform: Search specific parameter adapter

The guideline is, that the search specific filter adapter should be something that is easy for developers to maintain and understand.

The problem: The free text filter parameter is called “text“ in the LinkedEvents and “q“ in the Unified-Search. The UI has only 1 search field, which means that the parameter that it fills, needs to adapt to work in different searches.

The adapter should:

1. Pick the relevant URL search params; Exclude the ones that are not supported and pick the ones that could be used as they are or be transformed to a right form, so that they could be used as filters.
2. Clean the URL (as much as possible). The combined search has multiple search types and some parameters should persist even then when the search tab is not active.

The output (type `CombinedSearchAdapterOutput`) of the adapter is either `VenueSearchParams` or `EventSearchParams`.

---

An example of VenueSearchParams type, that could be used as an output for the venue search adapter:

```typescript
// NOTE: this is just an example, not an actual implementation
type VenueSearchParams = {
  q: string;
  ontologyWords: string[];
  orderBy: string;
};
```

if this would be the final active implementation of the type, the [CombinedSearchAdapterInput input type](#input-the-url-search-parameters) would transform to this like this:

1. `text` -> `q`.
2. `sportsCategories` -> `ontologyWords` (the adapter needs to also use the mapped values, not only the keys).
3. `orderBy` -> `orderBy`.

---

An example of EventSearchParams type, that could be used as an output for the event search adapter:

```typescript
// NOTE: this is just an example, not an actual implementation
type EventSearchParams = {
  text: string;
  keywords: string[];
  sort: string;
};
```

if this would be the final active implementation of the type, the [CombinedSearchAdapterInput input type](#input-the-url-search-parameters) would transform to this like this:

1. `text` -> `text`.
2. `sportsCategories` -> `keywords` (the adapter needs to also use the mapped values, not only the keys).
3. `courseOrderBy` -> `sort` & `eventOrderBy` -> `sort`.

---

#### The architecture

The URL acts as an input and a search adapter picks the params that the specific search needs and maps the query params to proper parameter field names. _In this visualization, the URL params are just some examples and may not be synced with the actual implementation._

```
┌──────────────────────────────┐     ┌────────────────────────────┐    ┌──────────────────────────────┐
| Input                        ├─────┤ Venue search adapter       ├────┤ Unified-Search               |
|==============================|     |============================|    |==============================|
| Given URL                    |     | Converted param            |    | Responds to                  |
| e.g /search?text=football&   |     | e.g /graphql?q=football    |    | param `q`                    |
| returnPath=/                 |     └────────────────────────────┘    └──────────────────────────────┘
|                              |
| Makes queries to multiple    |
| datasources                  |     ┌────────────────────────────┐    ┌──────────────────────────────┐
|                              |─────┤ Event search adapter       |────┤ LinkedEvents                 |
|                              |     |============================|    |==============================|
|                              |     | Converted param            |    | Responds to                  |
|                              |     | e.g /graphql?text=football |    | param `text`                 |
└──────────────────────────────┘     └────────────────────────────┘    └──────────────────────────────┘
```

CombinedSearchPage.tsx

```typescript
<SearchTabs initTab={initTab}>
  <CombinedSearchProvider>
    {/* The search form */}
    <SearchForm
      data-testid={searchContainerDataTestId}
      searchRoute={SEARCH_ROUTES.SEARCH}
      searchUtilities={null}
      korosBottom
      showTitle
      scrollToResultList={() => true}
    />

    {/* The search tabs, query sorters, search type switchers, etc. */}
    <SearchUtilities />

    {/* The Venue Search results */}
    <SearchTabs.Panel id="Venue">
      <VenueSearchPanel />
    </SearchTabs.Panel>

    {/* The General Event Search results */}
    <SearchTabs.Panel id={EventTypeId.General}>
      <EventSearchPanel eventType={EventTypeId.General} />
    </SearchTabs.Panel>

    {/* The Course Search results */}
    <SearchTabs.Panel id={EventTypeId.Course}>
      <EventSearchPanel eventType={EventTypeId.Course} />
    </SearchTabs.Panel>
  </CombinedSearchProvider>
</SearchTabs>
```

The `<SearchTabs/>` is there to provide a React-context / state for the tab system. The context of `<SearchTabs/>` tells

1. which tab should be the active one
2. how many search hits was found when a pre-search for the tab was done.

The `<SearchTabs/>` provider also updates the active `searchType` to the URL.

The `<CombinedSearchProvider/>` is there to have the formValues and the combined search variables stored in a React-context for each searchType. The `<CombinedSearchProvider/>` -component

1. reads the URL and uses the `CombinedSearchFormAdapter` to store the search form values in a context
2. provides the state mutator functions which can be used to update the form values in the context.
3. provides the search variables that can be used to query venues, courses and general events. The values are mapped with the `VenueSearchAdapter` and the `EventSearchAdapter` which both uses the `CombinedSearchFormAdapter` (the form values) as their input.

nikomakela committed:
feat(search): combined search adapters

LIIKUNTA-425

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr296.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
refactor(search): combined search adapters types usage

LIIKUNTA-425

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
feat(search): search specific variables

LIIKUNTA-425

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
feat(search): support all the (old) variables

LIIKUNTA-425

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
fix(search): the organisation search param should be undefinable

LIIKUNTA-425

nikomakela committed:
feat(search): combined search providers

LIIKUNTA-425

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
refactor: adapt the combined search to use new provider

LIIKUNTA-425

nikomakela committed:
fix(search): the venue query

LIIKUNTA-425.

While the venue search was still broken
and being fixed, it was found out that the
query variables are still some how
differing from the old ones.

The search variables are tested more properly. The default query variables are made to be
more like what they have been earlier.

nikomakela committed:
refactor(search): combined search url update functionalities

LIIKUNTA-425

nikomakela committed:
fix(search): variables in UnifiedSearchListQuery should be overridable

LIIKUNTA-425. 
The similar venues list query needs some
variables that cannot be fetched from 
current URL.

nikomakela committed:
refactor(search): rename components and relocate them inside the project

LIIKUNTA-425

nikomakela committed:
refactor(search): clean and comment the code

LIIKUNTA-425

nikomakela committed:
refactor(search): add a new controller hook for combined search

LIIKUNTA-425.

Remove the router usage totally from the
CombinedSearchFormAdapter, since it should
only map the URL search params to
search form input fields.

Add the removed features and tests to a new
React-hook that acts as a controller, that
decides when to navigate and to where.

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
docs(search): update the combined search readme

LIIKUNTA-425

nikomakela commented:
The code is now ready for reaview. Please also test that the Search form works as it should. There are lots of things to remember and lot for search form to do.

- [ ] Test that all the existing search features are still working after the combined search refactoring.

nikomakela committed:
fix(search): a missing type in the events search page

LIIKUNTA-425

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
style(search): fix css styles for combined search

LIIKUNTA-425

nikomakela committed:
docs(search): comment for text-parameter usage instead of allOngoingAnd

LIIKUNTA-425

nikomakela commented:
The ontologyTreeIds are currently populated wrong in the adapter. The adapter currently adds ontologyWordIds when it should add ontologyTreeIds. 

- [ ] Fix ontology tree ids queries
- [ ] Fix sportsCategories filter

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(search): sport locations filter

LIIKUNTA-425.

Instead of ontologyWordIds, the sportLocations
should be mapped to the ontologyTreeIds.

nikomakela committed:
fix(search): useLazyVenueSearchForTabCount still used the old variables

LIIKUNTA-425.

The variables of the venue search query
should be get from the combined search context.

nikomakela committed:
fix(search): map query should use the context search variables

LIIKUNTA-425

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
fix(search): default sport category usage

LIIKUNTA-425.

The default sport category should be used only when there are nothing else given.

nikomakela committed:
refactor(search): remove the unused useUnifiedSearchVariables

LIIKUNTA-425

nikomakela commented:
There is still an issue with the venue map search: The form is not initialized properly.

- [ ] Fix the venue map search form

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
fix(search): combined search provider for map view

LIIKUNTA-425.

The Combined search provider was not added
to the map search view, so the form was not
synced with the list version of it.

nikomakela committed:
fix(search): add code analysis improvments

LIIKUNTa-425

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders commented:
The sports categories filtering isn't working for events and hobbies:
![image](https://github.com/City-of-Helsinki/events-helsinki-monorepo/assets/77663720/b5f3ee4d-1253-4c75-891b-2fd0c7a276c3)

I have chosen "Tennis" as the sports category and that should filter it so that only tennis stuff is shown but the numbers don't change at all for hobbies and events.

![image](https://github.com/City-of-Helsinki/events-helsinki-monorepo/assets/77663720/bbb747e1-29b7-4d00-bc7f-0d7672458db5)
Here I have chosen "Hiihto" as the sports category and the numbers for hobbies and events are the same.

nikomakela committed:
fix(search): event search variables

LIIKUNTA-425

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela commented:
- [ ]  Add a support for internet-place filter that was in the old implementation

nikomakela commented:
- [ ] Divide the URL as an input in the architecture picture. Add the form adapter in it too.

melniiv reviewed (approved):
Code looks good

nikomakela commented:
There is actualyl stil lsome features like `getSearchFilters` that we should deprecate or even try to remove right now. They just makes everything inconsistent while they still are there, so I could try to remove then now. Lots of not so unnecessary code would be removed.

- [ ] Deprecate or remove the overlapping search form parameter handling features.

karisal-anders commented on apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts:
```diff
@@ -0,0 +1,82 @@
+    ] as UnifiedSearchLanguage;
+    this.q = input.text || initialVenueSearchAdapterValues.q;
+    this.ontologyTreeIds = this.getOntologyTreeIds(input);
+    this.orderByName = input.venueOrderBy
+      ? SortOrder.Ascending.toLowerCase().includes(input.venueOrderBy)
+        ? { order: SortOrder.Ascending }
+        : { order: SortOrder.Descending }
+      : initialVenueSearchAdapterValues.orderByName;
+    // TODO: this.orderByDistance = null; // input.venueOrderBy ? (SortOrder.Ascending.toLowerCase().includes(input.venueOrderBy) ? {order: SortOrder.Ascending} : {order: SortOrder.Descending}) : null;
```
Venue search sorting is not working (And this is the code that makes it so?),
tried it in https://liikunta-pr296.dev.hel.ninja/

nikomakela replied:
Should be fixed now. Also the distance sorter is now implemented and the event search sorters fixed.

nikomakela committed:
feat(search): support place and organisation filters

LIIKUNTA-425

Fix the search filters and refactor them
to use the new combined search context.

The old version supported publisher filter
and a place filter, so this should do that too.

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(search): reset form values

LIIKUNTA-425

nikomakela committed:
refactor(search): fixes to sorters and improvements to performance

LIIKUNTA-425

nikomakela committed:
docs(search): fixes to typos and improve architecture ascii image

LIIKUNTA-425

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(search): sort by location and event search sorting

LIIKUNTA-425

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(search): remove undefined keys from getQueryVariables

LIIKUNTA-425

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
test(search): fix the esmodule issues

LIIKUNTA-425

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
test: update the deprecated faker calls

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr296.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr296.dev.hel.ninja** :rocket::rocket::rocket:
